kedro-datasets/kedro_datasets/video/video_dataset.py

```diff
@@ -19,9 +19,10 @@
 class SlicedVideo:
     """A representation of slices of other video types"""
 
-    def __init__(self, video, slice_indexes):
+    def __init__(self, video, slice_indexes, metadata: Dict[str, Any] = None):
```

Comment:
The `SlicedVideo` class isn't a dataset, so it doesn't make sense to add the metadata here.